### DISCUSSION TITLE
How to add UserId of logged in user to a Patient table while adding new patient record in serenity?

### DISCUSSION THREAD

**Discussion by imhemayatsangin** [serenity-is/Serenity#5831] [Help] [answered]
I have a Patient Table and in this table i have a column UserId where i store user id of a logged in user automatically in the background when i insert Patient data. I tried some methods shown here but i couldn't get any solutions.Please can anybody help me how to solve this issue. Im using Serene latest verson .net core.

**Comment by edwardch** [comment, marked as answer]
Have a look at LoggingRow or ILoggingRow. If you only care about inserts then you can just use IInsertLogRow.

**Reply by imhemayatsangin** [reply]
How to use this? i checked but if you can guide me exactly how to do it, Please kindly share the whole code my code for UserId(which is a foreign key) in the PatientRow.cs is as follows:

        [DisplayName("User"), NotNull, ForeignKey("[dbo].[Users]", "UserId"), LeftJoin("jUser"), TextualField("UserUsername")]
        public Int32? UserId
        {
            get => fields.UserId[this];
            set => fields.UserId[this] = value;
        }

        [DisplayName("User Username"), Expression("jUser.[Username]")]
        public String UserUsername
        {
            get => fields.UserUsername[this];
            set => fields.UserUsername[this] = value;
        }

**Reply by luongnv6** [reply]
Also you can also use this way, if you don't want to use LoggingRow 
```C#
        private class MySaveHandler : SaveRequestHandler<MyRow>
        {
            protected override void BeforeSave()
            {
                base.BeforeSave();

                var user = Authorization.UserDefinition;

                if (IsCreate)
                {
                    Row.InsertById = Convert.ToInt32(user.Id);
                    Row.InsertDate = DateTime.Now;
                }
                else
                {
                    Row.UpdateById = Convert.ToInt32(user.Id);
                    Row.UpdateDate = DateTime.Now;
                }
            }
        }
```

**Reply by aliffauzi34** [reply]
> Anda juga dapat menggunakan cara ini, jika Anda tidak ingin menggunakan LoggingRow
> 
> ```cs
>          kelas  pribadi MySaveHandler : SaveRequestHandler < MyRow >
>         {
>             protected  override  void  BeforeSave ()
>             {
>                 dasar . SebelumSimpan ();
> 
>                 var  pengguna  =  Otorisasi . Definisi Pengguna ;
> 
>                 jika ( IsCreate )
>                 {
>                     Baris . InsertById  =  Konversi . ToInt32 ( pengguna . Id );
>                     Baris . InsertDate  =  TanggalWaktu . Sekarang ;
>                 }
>                 lain
>                 {
>                     Baris . UpdateById  =  Konversi . ToInt32 ( pengguna . Id );
>                     Baris . UpdateDate  =  TanggalWaktu . Sekarang ;
>                 }
>             }
>         }
> ```

I tried with this code but got error like this
<img width="481" alt="Capture" src="https://user-images.githubusercontent.com/56574544/124060602-97e04780-da57-11eb-8ef7-769546a752aa.PNG">

**Reply by imhemayatsangin** [reply]
Thanks luongnv6

**Reply by imhemayatsangin** [reply]
Thanks luongnv6

**Comment by imhemayatsangin** [comment]
An easiest way to add and save UserID of the logged in user automatically with a particular entry in Serenity asp.net core version is described in this YouTube video with an Example:

[https://www.youtube.com/watch?v=r2LenNuucLg](https://www.youtube.com/watch?v=r2LenNuucLg)
